src/ts/waterfall/details-overlay/html-details-body.ts

```diff
@@ -1,146 +1,45 @@
-}
-
-function makeGeneralTab(generalData: KvTuple[], entry: WaterfallEntry) {
-  let content = makeDefinitionList(generalData);
-  if (entry.indicators.length === 0) {
-    return makeTab(content, true);
-  }
-  let general = `<h2>General</h2>
-    <dl>${content}<dl>`;
-  content = "";
-
-  // Make indicator sections
-  let errors = entry.indicators
-    .filter((i) => i.type === "error")
-    .map((i) => [i.title, i.description] as KvTuple);
-  let warnings = entry.indicators
-    .filter((i) => i.type === "warning")
-    .map((i) => [i.title, i.description] as KvTuple);
-  // all others
-  let info = entry.indicators
-    .filter((i) => i.type !== "error" && i.type !== "warning")
-    .map((i) => [i.title, i.description] as KvTuple);
-
-  if (errors.length > 0) {
-    content += `<h2 class="no-boder">Error${errors.length > 1 ? "s" : ""}</h2>
-    <dl>${makeDefinitionList(errors)}</dl>`;
-  }
-  if (warnings.length > 0) {
-    content += `<h2 class="no-boder">Warning${warnings.length > 1 ? "s" : ""}</h2>
-    <dl>${makeDefinitionList(warnings)}</dl>`;
-  }
-  if (info.length > 0) {
-    content += `<h2 class="no-boder">Info</h2>
-    <dl>${makeDefinitionList(info)}</dl>`;
-  }
-  return makeTab(content + general, false);
-}
-
-function makeTabBtn(name: string, tab: string) {
-  return !!tab ? `<li><button class="tab-button">${name}</button></li>` : "";
-}
-
-function escapeHtml(unsafe) {
-  return unsafe
-    .replace(/&/g, "&amp;")
-    .replace(/</g, "&lt;")
-    .replace(/>/g, "&gt;")
-    .replace(/"/g, "&quot;")
-    .replace(/'/g, "&#039;");
-}
-
-export function createDetailsBody(requestID: number, entry: WaterfallEntry, accordeonHeight: number) {
+export function createDetailsBody(requestID: number, detailsHeight: number, entry: WaterfallEntry) {
 
   let html = document.createElement("html") as HTMLHtmlElement;
   let body = document.createElement("body");
   body.setAttribute("xmlns", "http://www.w3.org/1999/xhtml");
   html.setAttributeNS("http://www.w3.org/2000/xmlns/", "xmlns", "http://www.w3.org/2000/xmlns/");
 
-  const tabsData = getKeys(requestID, entry);
-  const generalTab = makeGeneralTab(tabsData.general, entry);
-  const timingsTab = makeTab(makeDefinitionList(tabsData.timings, true));
-  const requestDl = makeDefinitionList(tabsData.request);
+  const tabMenu = entry.tabs.map((t) => {
+    return `<li><button class="tab-button">${t.title}</button></li>`;
+  }).join("\n");
 
-  const requestHeadersDl = makeDefinitionList(tabsData.requestHeaders);
-  const responseDl = makeDefinitionList(tabsData.response);
-  const responseHeadersDl = makeDefinitionList(tabsData.responseHeaders);
-  const imgTab = makeImgTab(accordeonHeight, entry);
+  const tabBody = entry.tabs.map((t) => {
+    let cssClasses = "tab";
+    if (t.tabClass) {
+      cssClasses += ` ${t.tabClass}`;
+    }
+    let content = "";
+    if (t.content) {
+      content = t.content;
+    } else if (typeof t.renderContent === "function") {
```

Comment:
for lazy evaluation of tabs - cuts of >10ms, for HARs with content, since the replcement of large content takes a while.
Also used for image tab to get the runtime `detailsHeight` variable.